datalad/core/distributed/clone.py

```diff
@@ -581,21 +581,141 @@ def postclonecfg_ria(ds, props):
+        # create-sibling-ria and enable try enabling that one.
+        lgr.debug("Found no autoenabled ORA special remote. Trying to look it "
+                  "up in source config ...")
+
+        # First figure whether we cloned via SSH, HTTP or local path and then
+        # get that config file the same way:
+        config_content = None
+        scheme = props['giturl'].split(':')[0]
+        if scheme == 'http':
+
+            # Note: Following outcommented code was proof how to check via HTTP.
+            #       However, for now we only want to annexenable origin in case
+            #       of HTTP, not the special remote. Hence it's uuid isn't
+            #       actually required. (see gh-4410)
+
+            # import requests
+            # try:
+            #     response = requests.get("{}{}config".format(
+            #         props['giturl'],
+            #         '/' if not props['giturl'].endswith('/') else '')
+            #     )
+            #     config_content = response.text
+            # except requests.RequestException as e:
+            #     lgr.debug("Failed to get config file from source:\n%s",
+            #               exc_str(e))
+
+            lgr.debug("No ORA reconfiguration for HTTP implemented yet")
+
+        elif scheme == 'ssh':
+
+            # TODO: switch the following to proper command abstraction:
+            from datalad.distributed.ora_remote import SSHRemoteIO, RIARemoteError
+            # SSHRemoteIO ignores the path part ATM. No remote CWD! (To be
+            # changed with command abstractions). So we need to get that part to
+            # have a valid path to origin's config file:
+            from datalad.support.network import URL
+            from datalad.utils import PurePosixPath
+            cfg_path = PurePosixPath(URL(props['giturl']).path) / 'config'
+            op = SSHRemoteIO(props['giturl'])
+            try:
+                config_content = op.read_file(cfg_path)
+            except RIARemoteError as e:
+                lgr.debug("Failed to get config file from source: %s",
+                          e)
+
+        elif scheme == 'file':
+
+            # TODO: switch the following to proper command abstraction:
+            from datalad.distributed.ora_remote import LocalIO, RIARemoteError
+            op = LocalIO()
+            from datalad.support.network import URL
+            cfg_path = Path(URL(props['giturl']).localpath) / 'config'
+            try:
+                config_content = op.read_file(cfg_path)
+            except (RIARemoteError, OSError) as e:
+                lgr.debug("Failed to get config file from source:\n%s",
+                          e)
+        else:
+            lgr.debug("Unknown URL-Scheme in %s. Can currently handle SSH or "
+                      "FILE scheme URLs.", props['source'])
+
+        # 3. And read it
+        org_uuid = None
+        if config_content:
+            from datalad.utils import make_tempfile
```

Comment:
no reason to delay imports from core modules such as `datalad.utils` (and probably such as `.support.network`) all the way to the point of use.  Those modules would get imported anyways somewhere else in the code and should  not be taxing on performance.